### DISCUSSION TITLE
polygon labeling issue

### DISCUSSION THREAD

**Discussion by 4tler** [mapsforge/vtm#850] [Q&A] [answered]
Hello, I've been working on creating my own theme for mapsforge vtm vector data for a short time. I noticed that polygons are unfortunately labeled multiple times when you increase the zoom level. I have tried this with the example app and the themes integrated there and the problem remains. Is there a workaround for this? 

my theme:
![image](https://user-images.githubusercontent.com/74738662/121150805-204a3e80-c844-11eb-9c6d-e0c3d633b5d7.png)
osmarender theme:
![image](https://user-images.githubusercontent.com/74738662/121150947-3eb03a00-c844-11eb-9497-bc10fbcc2240.png)

**Comment by devemux86** [comment]
_Please write the coordinates of your examples._

Are you sure the campings are not nodes?

For polygon labels you can try VTM "area-size" parameter of  caption / text rules.
See in xml schema [[1](https://github.com/mapsforge/vtm/blob/master/resources/rendertheme.xsd#L157-L158), [2](https://github.com/mapsforge/vtm/blob/master/resources/rendertheme.xsd#L228-L229)] and an example in [default.xml](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/default.xml#L466).

There is also the [Parameters](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/utils/Parameters.java) class with static flags that affect various things in the library.

**Reply by 4tler** [reply]
Coordinates: [51.937050567024656,13.89522135257721](https://www.openstreetmap.org/?mlat=51.93758860230446&mlon=13.89445960521698#map=18/51.93759/13.89446)

We will check the VTM area size parameter but the problem is also in the standard themes.

**Comment by devemux86** [comment]
I cannot reproduce what you report with brandenburg.map from [Mapsforge server](https://download.mapsforge.org/maps/v5/), VTM themes and vtm-playground.

You can add or increase the "area-size" parameter if want fewer labels.

![VTM](https://user-images.githubusercontent.com/3484020/121164673-66a99880-c858-11eb-954c-2d842c834df7.png)

**Reply by 4tler** [reply]
i think the standard mapsforge files don't support this. because they don't render e.g. camping sites as areas but only as poi. i added some things to my tag mapping to really have all camping sites in the map. So nodes and areas.

here is an example .map file produced with my tag-mapping.xml

[SpreeNav_Spreewald.zip](https://github.com/mapsforge/vtm/files/6615599/SpreeNav_Spreewald.zip)

by the way, is there a option where i can find support for Cruiser desktop (WIN)? Because the Cruiser App works perfect but the Cruiser-gl crash with every start. Testet with 2 systems.

**Comment by devemux86** [comment, marked as answer]
Use the [Parameters.POLY_CENTROID](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/utils/Parameters.java#L56-L60) flag.

As mentioned in the [comments](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java#L1016-L1022),
if map files are not created with [global](https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md#advanced-options-only-use-when-you-know-what-you-are-doing) or [local](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml) label-position,
you can force that calculation when reading the map files.

**Reply by 4tler** [reply]
we will check this and report...thanks

**Reply by 4tler** [reply]
thank's, that's the problem and the solution.

**Comment by devemux86** [comment]
It is impossible to examine everyone's code and examples.
Reports need to be reproduced with official samples / maps.

There are cases with official maps with multiple polygon labels.
Some users may need it, specially with large polygons (to not be empty).
Everyone else can use the [Parameters.POLY_CENTROID](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/utils/Parameters.java#L56-L60) flag.
